Disable test_dmabuf_import on the same conditions as the functions it uses
Patch by Simon Branford (University of Birmingham)
--- testsuite/gdk/dmabuftexture.c.orig	2025-10-13 15:07:06.809685000 +0100
+++ testsuite/gdk/dmabuftexture.c	2025-10-13 15:14:14.446178000 +0100
@@ -175,6 +175,7 @@
  * then import it into another GL context, download
  * the resulting texture, and compare it to the original.
  */
+#if defined(HAVE_DMABUF) && defined (HAVE_EGL)
 static void
 test_dmabuf_import (void)
 {
@@ -257,6 +258,7 @@
   g_object_unref (context);
   g_object_unref (context2);
 }
+#endif
 
 int
 main (int argc, char *argv[])
@@ -264,7 +266,9 @@
   gtk_test_init (&argc, &argv, NULL);
 
   g_test_add_func ("/dmabuf/export", test_dmabuf_export);
+#if defined(HAVE_DMABUF) && defined (HAVE_EGL)
   g_test_add_func ("/dmabuf/import", test_dmabuf_import);
+#endif
 
   return g_test_run ();
 }
--- testsuite/gdk/dmabuf-support.c.orig	2025-10-13 15:39:32.922287913 +0100
+++ testsuite/gdk/dmabuf-support.c	2025-10-13 15:40:45.657530286 +0100
@@ -4,6 +4,7 @@
 #include "gdk/gdkdmabuffourccprivate.h"
 #include "udmabuf.h"
 
+#if defined(HAVE_DMABUF) && defined (HAVE_EGL)
 static void
 test_dmabuf_no_gpu (void)
 {
@@ -55,13 +56,16 @@
 
   g_object_unref (texture);
 }
+#endif
 
 int
 main (int argc, char *argv[])
 {
   gtk_test_init (&argc, &argv, NULL);
 
+#if defined(HAVE_DMABUF) && defined (HAVE_EGL)
   g_test_add_func ("/dmabuf/no-gpu", test_dmabuf_no_gpu);
+#endif
 
   return g_test_run ();
 }
